cl/alerts/models.py

```diff
@@ -105,17 +106,28 @@ class DocketAlert(models.Model):
     )
     objects = DocketAlertManager()
 
+    __original_alert_type = None
+
     class Meta:
         unique_together = ("docket", "user")
 
     def __str__(self) -> str:
         return f"{self.pk}: {self.docket_id}"
 
+    def __init__(self, *args, **kwargs):
+        super(DocketAlert, self).__init__(*args, **kwargs)
+        self.__original_alert_type = self.alert_type
+
     def save(self, *args, **kwargs):
         """Ensure we get a token when we save the first time."""
         if self.pk is None:
             self.secret_key = get_random_string(length=40)
+
+        if self.alert_type != self.__original_alert_type:
+            self.date_created = now()
+
         super(DocketAlert, self).save(*args, **kwargs)
+        self.__original_alert_type = self.alert_type
 
```

Comment:
This is clever, but I don't like it because it breaks the `update()` method, and it makes it so that the date_created field is longer trustworthy. If we must, we can add another column for this purpose, or maybe could we use date_modified instead?

Reply:
Got it, yeah I think we could add a new column like `date_modified` (we don't have it right now in the `DocketAlert` model) so we can save the date when the docket alert `alert_type` is toggled. 
However, I think we'll need to continue using the custom `save()` method instead of `update()` or the `save()` method without the hack above and a `post_save` signal, so we can update the `date_modified` only when the `alert_type` changes so that we can use it to calculate the docket alert age. 
Does it make sense?

Reply:
Hm, well, I'm shocked we don't have a date_modified field on here. Let's fix that by having it inherit the usual class that gets us that field (assuming that doesn't mess up the existing date_created field; I don't think it will). 

As for updating the save function, I'd much rather have a date_enabled field than do that spooky stuff at a distance, or just rely on the date_modified field, since it shouldn't get updated too often anyway?

Reply:
Thanks, I'll check if we can add the `date_modified` inheriting from `AbstractDateTimeModel`.

So doing that we could rely on the `auto_now` property of `date_modified` so it's updated every time the `save()` is called, no need for hacks in the `save()` method. I've confirmed that when updating the `date_last_hit` the `date_modified` won't be updated since it uses the `update()` method.

Reply:
Sounds good! I still can't get over the fact that `update()` doesn't handle auto_now fields. Come on, Django!

Reply:
This is changed, I added `date_modified` inheriting from `AbstractDateTimeModel` and now using `date_modified` to keep track of old docket alerts.